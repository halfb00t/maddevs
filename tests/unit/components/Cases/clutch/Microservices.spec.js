import { render } from '@testing-library/vue'
import { createLocalVue } from '@vue/test-utils'
import Microservices from '@/components/Cases/clutch/Microservices'
// eslint-disable-next-line import/order
import lazyLoad from 'nuxt-lazy-load/lib/module'

const mocks = {
  $getMediaFromS3: () => 'img.jpg',
}

const localVue = createLocalVue()
localVue.directive('lazy-load', lazyLoad)
jest.mock('nuxt-lazy-load/lib/module')

describe('Microservices component', () => {
  it('should render correctly', () => {
    const { container } = render(Microservices, {
      mocks,
      localVue,
    })

    expect(container).toMatchSnapshot()
  })
})
