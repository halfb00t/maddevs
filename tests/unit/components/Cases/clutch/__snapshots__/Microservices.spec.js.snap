// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Microservices component should render correctly 1`] = `
<div>
  <section
    class="p-96_bottom p-media-p-48_bottom"
  >
    <div
      class="container_regular m-96_bottom media-m-48_bottom"
    >
      <h2
        class="case_title_h2 m-12_bottom media-m-12_bottom"
      >
        
      Switched to the microservices architecture
    
      </h2>
       
      <p
        class="case_paragraph"
      >
        
      Although Clutch.co’s monolithic architecture was cost-effective at the early development stages, it couldn’t
      sustain Сlutch’s continuously increasing website traffic and new registrations. So Mad Devs stepped in to build
      a new microservices architecture from scratch on Go and Python.
    
      </p>
    </div>
     
    <div
      class="container_full background-color-blue-grey p-100_top p-100_bottom media-p-41_top media-p-41_bottom"
    >
      <div
        class="container_regular"
      >
        <div
          class="case_cards-group"
        >
          <div
            class="case_card case_card"
          >
            <h2
              class="case_title_h2 case_card-title m-21_bottom media-m-10_bottom"
            >
              
            Integrated single sign-on (SSO)
          
            </h2>
             
            <p
              class="case_paragraph case_card-text m-24_bottom media-m-2_bottom"
            >
              
            The integration of SSO challenged our team with the need to display different web designs based on the URL
            that the user initially accessed. We made all services talk to each other via a remote procedure call
            system, and the SSO functionality was designed to quickly take the user to the desired website in a
            user-friendly way.
          
            </p>
          </div>
           
          <div
            class="case_card case_card"
          >
            <picture
              class="case_card-img"
            >
              <source
                class="image"
                data-srcset="img.jpg ,img.jpg"
                type="image/webp"
              />
               
              <img
                alt="SSO"
                class="image"
                data-src="img.jpg"
                data-srcset="img.jpg"
                data-testid="test-picture-img"
                height="695"
                width="359"
              />
            </picture>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
